api/src/main/java/org/openmrs/module/appointments/service/impl/SMSService.java

```diff
@@ -0,0 +1,116 @@
+        this.appointmentVisitLocation = appointmentVisitLocation;
+    }
+
+    public String getAppointmentMessage(String name, String familyName, String id, Date appointmentDate, String service, List<String> providers, Location location) {
+        String smsTemplate = Context.getAdministrationService().getGlobalProperty(APPOINMTMENT_REMINDER_SMS_TEMPLATE);
+        String smsTemplateNoProvider = Context.getAdministrationService().getGlobalProperty(APPOINMTMENT_REMINDER_SMS_TEMPLATE_NO_PROVIDER);
+        String smsTimeZone = Context.getMessageSourceService().getMessage(SMS_TIMEZONE, null, new Locale("en"));
+        String smsDateFormat = Context.getMessageSourceService().getMessage(SMS_DATEFORMAT, null, new Locale("en"));
+        String date =convertUTCToGivenFormat(appointmentDate, smsDateFormat, smsTimeZone);
+        String helpdeskNumber = Context.getAdministrationService().getGlobalPropertyObject("clinic.helpDeskNumber").getPropertyValue();
+        LocationTag abc = Context.getLocationService().getLocationTagByName("Visit Location");
+        List<Location> locations = Context.getLocationService().getLocationsHavingAnyTag(Collections.singletonList(abc));
+        String clinicName = (abc != null && !locations.isEmpty()) ? locations.get(0).getName() : "xxxxx";
+        if (location != null) {
+            String cName = appointmentVisitLocation.getClinicName(location.getUuid());
+            if (StringUtils.isNotEmpty(cName)) {
+                clinicName = cName;
+            }
+        }
+        List<Object> arguments = new ArrayList<>();
+        arguments.add(name + " " + familyName);
+        arguments.add(id);
+        arguments.add(date);
+        arguments.add(org.springframework.util.StringUtils.collectionToCommaDelimitedString(providers));
+        arguments.add(service);
+        arguments.add(clinicName);
+        arguments.add(helpdeskNumber);
+
+        String message;
+        if (StringUtils.isBlank(smsTemplate)) {
+            if (providers.isEmpty()) {
+                arguments.remove(3);
+                message = Context.getMessageSourceService().getMessage(APPOINMTMENT_REMINDER_SMS_TEMPLATE_NO_PROVIDER, arguments.toArray(), new Locale("en")).replace("\\n", System.lineSeparator());
+            } else {
+                message = Context.getMessageSourceService().getMessage(APPOINMTMENT_REMINDER_SMS_TEMPLATE, arguments.toArray(), new Locale("en")).replace("\\n", System.lineSeparator());
+            }
+        } else {
+            if (providers.isEmpty()) {
+                arguments.remove(3);
+                message=new MessageFormat(smsTemplateNoProvider).format(arguments.toArray()).replace("\\n", System.lineSeparator());
+            } else {
+                message =new MessageFormat(smsTemplate).format(arguments.toArray()).replace("\\n", System.lineSeparator());
+            }
+        }
+        return message;
+    }
+
+    public Object sendSMS(String phoneNumber, String message) {
```

Comment:
Why does this method return an Object, instead of a boolean or String, or Status object?